cns/cnireconciler/statefile.go

```diff
@@ -0,0 +1,41 @@
+package cnireconciler
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+
+	"github.com/Azure/azure-container-networking/cns/logger"
+)
+
+// WriteObjectToCNIStatefile checks for a file at the CNI statefile path,
+// and checks if it is empty. If it is empty, writes an empty JSON object to
+// it so older CNI can execute. Does nothing and returns no error if the
+// file does not exist.
+//
+// This is a hack to get older CNI to run when CNS has mounted the statefile
+// path, but the statefile wasn't written by CNI yet. Kubelet will stub an
+// empty file on the host filesystem, crashing older CNI because it doesn't know
+// how to handle empty statefiles.
+func WriteObjectToCNIStatefile() error {
+	filename := "/var/run/azure-vnet.json"
```

Comment:
Hi @rbtr 👋 what's the best way I can make this different for OS windows? 

I was thinking about making a `statefile_windows.go` with this same code but different filename, that seems to copy alot however.

Any ideas? I just want to defer to you since you know alot more about go than I do, I can only think about an if statement if the OS is windows, just wondering if that's enough

Reply:
this was a hack for older CNIs, pre 1.4.7 or so; so if you're adding support to Windows now then this might not even be necessary.
if it is necessary, I would expect the only thing that's different on Windows is the filepath, right? So that should be injected from an early OS check all the way down in to this function. OR, probable more correctly, this shouldn't be a hardcoded const and should come from env or config.